src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -228,7 +250,7 @@ public ZooKeeperSaslClient getSaslClient() {
      * the public methods will not be exposed as part of the ZooKeeper client
      * API.
      */
-    static class ZKWatchManager implements ClientWatchManager {
+    class ZKWatchManager implements ClientWatchManager {
```

Comment:
The changes i suggested, i already have done it and it is in the latest commit. ZKWatchManager  had default constructor, I added parameterized constructor in it as 
ZKWatchManager(boolean disableAutoWatchReset) {
            this.disableAutoWatchReset = disableAutoWatchReset;
       }
and reverted it to static class as it was earlier before this patch change.